### PR TITLE
quote-and-reply: Set topic input when appropriate; focus message input

### PR DESCRIPTION
**Pull request by chrisbobbe**
Fixes: #5718

### CODE REVIEW

File: src/compose/ComposeBox.js [hunk 1]
```diff
@@ -445,6 +449,7 @@ const ComposeBox: React$AbstractComponent<Props, ImperativeHandle> = forwardRef(
       } finally {
         setActiveQuoteAndRepliesCount(v => v - 1);
         activeInvocations.current = activeInvocations.current.filter(x => x !== serialNumber);
+        messageInputRef.current?.focus();
```
Comment (gnprice) [inline src/compose/ComposeBox.js:452]: What's the reasoning for putting this step in the `finally` block?

I think it should instead go at the end of the `try` block. The existing job of this `finally` block is purely about making sure that the `setActiveQuoteAndRepliesCount(v => v + 1)` and `activeInvocations.current.push` at the top of the function always get closed eventually, regardless of whether the invocation succeeds or fails. If it does fail, I think it's probably not helpful to go and focus the input — we haven't successfully set up something for the user to type their reply after.

**Review by gnprice — COMMENTED** [review]
This plan sounds good. One comment below.

File: src/compose/ComposeBox.js (hunk 1 above)
Comment (chrisbobbe) [reply src/compose/ComposeBox.js:452]: Oh interesting, yeah, I agree it seems better to put it at the end of the `try`. Will test to make sure that doesn't break anything.

Comment (chrisbobbe) [reply src/compose/ComposeBox.js:452]: Yep, that seems to work fine. Thanks for the catch!

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

**Comment by gnprice** [comment]
Thanks! Looks good; merging.

**Pull request merged**